ADCM-1858 UI spec and tests refactoring (#921)
arenadata/adcm: spec/ui/actions.rst

#########
Use Cases
#########

.. _ui_usecases_login:

*****
Login
*****

There is a number of scenarios related to login forms of ADCM.

.. note::
   Upgrade of ADCM and related events is out of scope for this use case.

User/password login
===================

Just a simple use case of auth for user with username and password.

**Goal**: successful login

1. Go to ADCM
2. Fill *Login* on :ref:`ui_elements_forms_login`
3. Fill *Password* on :ref:`ui_elements_forms_login`
4. Press button *Login* on :ref:`ui_elements_forms_login`

Exceptions:

> 2. User already logged to ADCM

Reviewer: ```suggestion
2. User has been already logged in to ADCM
```

Reviewer: What about the case when the user has been already logged in but navigates to http://localhost:8004/login?
The case above is a bit weird since in this case for some reason user gets logged out without any warnings

    a. There is no :ref:`ui_elements_forms_login`

2. User does not fill *Login* on :ref:`ui_elements_forms_login`
    a. Button *Login* on :ref:`ui_elements_forms_login` is inactive (no way to click it)

2. User does not fill *Password* on :ref:`ui_elements_forms_login`
    a. Button *Login* on :ref:`ui_elements_forms_login` is inactive (no way to click it)

GAuth2 login
============

.. note::
   TODO: Add later as separate story.

> 

Reviewer: Logout?

Author: Gauth

Author: It is mention in a header

Reviewer: I mean that the Logout scenario should be also described. Especially situations when you navigate to the /login page and get logged out with no additional notifications.

Pre-Conditions
--------------

:term:`End User` has ADCM and completed the use case :ref:`ui_usecases_login`.

Post-Conditions
---------------

:term:`End User` was able to view the list of clusters and information on them.

“Used” Use Cases
----------------

* :ref:`ui_usecases_cluster_service_list_view`

Flow of Events
--------------

1. :term:`End User` goes to the "Cluster" tab of the main menu of the :ref:`ui_templates_common` template.
2. :term:`End User` sees a list of clusters and information on them.

User Interface
--------------

The description of the UI is available by clicking :ref:`ui_templates_clusters`.

.. _ui_usecases_cluster_create:

Cluster create
==============

The way user create a cluster in UI.

Actors
------

* :term:`End User`

User Value
----------

That is the only way to create cluster.

Pre-Conditions
--------------

:term:`End User` has ADCM and completed the following use cases:

* :ref:`ui_usecases_login`
* :ref:`ui_usecases_cluster_list_view`

Flow of Events
--------------

1. :term:`End User` clicks "Create Cluster" button in :ref:`ui_form_dialogs_create_cluster`

> 2. :term:`End User` selects bundle from "Bundle" selector in  :ref:`ui_form_dialogs_create_cluster`

Reviewer: Besides selecting the uploaded bundle we can also upload a new one right from this dialog menu.

Author: Not now

3. :term:`End User` selects version from "Version" selector in  :ref:`ui_form_dialogs_create_cluster`
4. :term:`End User` fills "Cluster name" field in :ref:`ui_form_dialogs_create_cluster`
5. :term:`End User` fills "Description" field in :ref:`ui_form_dialogs_create_cluster`
6. :term:`End User` clicks "Create" button in :ref:`ui_form_dialogs_create_cluster`

Post-Conditions
---------------

* A Cluster has been created.
* :ref:`ui_form_dialogs_create_cluster` has beeen closed.

> 

Reviewer: Add services to cluster?

Author: What for? This is the end of this scenario, because scenario was called "create cluster"

Reviewer: Fine. Then it should be a standalone scenario.

Cluster service list view
=========================

> Use case in which the user can view the list of added clusters and information on them.

Reviewer: ```suggestion
Use case in which the user can view the list of added services for the given cluster and information on them.
```

Actors
------

* :term:`End User`

User Value
----------

> This use case will allow the :term:`End User` view the list of services included to cluster.

Reviewer: ```suggestion
This use case will allow the :term:`End User` to view the list of services included in the cluster.
```

--------------

:term:`End User` has ADCM and completed the following use cases:

* :ref:`ui_usecases_login`
* :ref:`ui_usecases_cluster_list_view`

Post-Conditions
---------------

* :term:`End User` was able to view the list of cluster hosts.

“Used” Use Cases
----------------

Flow of Events
--------------

1. :term:`End User` selects the required cluster by clicking on it in the list of clusters.
2. :term:`End User` goes to the "Hosts" section in the Left Menu.
3. :term:`End User` sees a list of cluster hosts.

User Interface
--------------

List of host described in :ref:`ui_templates_cluster_host_list_view`.

*************
Host Provider
*************

.. _ui_usecases_create_hostprovider:

Host Provider Create
====================

The way user create a Host Provider in UI.

Actors
------

* :term:`End User`

User Value
----------

That is the only way to create Host Provider.

Pre-Conditions
--------------

:term:`End User` has ADCM and completed the following use cases:

* :ref:`ui_usecases_login`
* :ref:`ui_usecases_hostprovider_list_view`

Flow of Events
--------------

1. :term:`End User` clicks "Create provider" button in :ref:`ui_form_dialogs_create_hostprovider`

> 2. :term:`End User` selects bundle from "Bundle" selector in  :ref:`ui_form_dialogs_create_hostprovider`

Reviewer: Samer here about bundle upload from this menu

Author: Don't what to do this in this scenario.
